lib/ansible/modules/web_infrastructure/ansible_tower/tower_credential.py

```diff
@@ -165,10 +165,14 @@
     ssh_key_data: "{{ lookup('file', '/tmp/id_rsa') }}"
     ssh_key_unlock: "passphrase"
 
+- name: Fetch private key
+  slurp:
+    src: '$HOME/.ssh/aws-private.pem'
+  register: aws_ssh_key
```

Comment:
reading the docs, I don't quite understand how slurp avoids the problem of looking up on the managed node vs. the control node, whereas lookup doesn't.

Reply:
`/home/{{ansible_user}}/.ssh/aws-private.pe` is a path on managed node (`ansible_user` is user on managed node)
- `file` lookup is a Jinja filter, template are [rendered on the controller](https://docs.ansible.com/ansible/latest/user_guide/playbooks_templating.html) (`ansible_user` could not exists here)
- `slurp` is an Ansible module executed on managed host

Reply:
oh got it, I had it backwards. The goal here is to create a credential from data on the managed node.